code/__HELPERS/_logging.dm

```diff
@@ -58,26 +58,32 @@ GLOBAL_PROTECT(log_end)
 		rustg_log_write(GLOB.world_game_log, "ACCESS OUT: [message][GLOB.log_end]")
 
 /proc/log_say(text, mob/speaker)
+	check_for_troll("Say", text, speaker)
 	if(config.log_say)
 		rustg_log_write(GLOB.world_game_log, "SAY: [speaker.simple_info_line()]: [html_decode(text)][GLOB.log_end]")
 
 /proc/log_whisper(text, mob/speaker)
+	check_for_troll("Whisper", text, speaker)
 	if(config.log_whisper)
 		rustg_log_write(GLOB.world_game_log, "WHISPER: [speaker.simple_info_line()]: [html_decode(text)][GLOB.log_end]")
 
 /proc/log_ooc(text, client/user)
+	check_for_troll("OOC", text, user)
 	if(config.log_ooc)
 		rustg_log_write(GLOB.world_game_log, "OOC: [user.simple_info_line()]: [html_decode(text)][GLOB.log_end]")
 
 /proc/log_aooc(text, client/user)
+	check_for_troll("AOOC", text, user)
```

Comment:
Players dont speak in this. Only admins do.

Reply:
Thanks! Removed this one.